plugins/inputs/opcua/opcua_test.go

```diff
@@ -185,74 +187,49 @@ private_key = "/etc/telegraf/key.pem"
 auth_method = "Anonymous"
 username = ""
 password = ""
-nodes = [
-  {name="name", namespace="1", identifier_type="s", identifier="one", tags=[["tag0", "val0"]]},
-  {name="name2", namespace="2", identifier_type="s", identifier="two", tags=[["tag0", "val0"], ["tag00", "val00"]]},
-]
+
+[[inputs.opcua.nodes]]
+  name = "name"
+  namespace = "1"
+  identifier_type = "s"
+  identifier="one"
+  tags=[["tag0", "val0"]]
+
+[[inputs.opcua.nodes]]
+  name="name2"
+  namespace="2"
+  identifier_type="s"
+  identifier="two" 
+  tags=[["tag0", "val0"], ["tag00", "val00"]]
+  default_tags = {tag6 = "val6"}
+
 [[inputs.opcua.group]]
 name = "foo"
 namespace = "3"
 identifier_type = "i"
 tags = [["tag1", "val1"], ["tag2", "val2"]]
 nodes = [{name="name3", identifier="3000", tags=[["tag3", "val3"]]}]
+
 [[inputs.opcua.group]]
 name = "bar"
 namespace = "0"
 identifier_type = "i"
 tags = [["tag1", "val1"], ["tag2", "val2"]]
-nodes = [{name="name4", identifier="4000", tags=[["tag1", "override"]]}]
+[[inputs.opcua.group.nodes]]
+  name = "name4"
+  identifier = "4000"
+  tags=[["tag4", "val4"]]
+  default_tags = { tag1 = "override" }
+
+[[inputs.opcua.group.nodes]]
+  name = "name5"
+  identifier = "4001"
 
 [inputs.opcua.workarounds]
 additional_valid_status_codes = ["0xC0"]
-`
-
-	c := config.NewConfig()
-	err := c.LoadConfigData([]byte(toml))
-	require.NoError(t, err)
-
-	require.Len(t, c.Inputs, 1)
-
-	o, ok := c.Inputs[0].Input.(*OpcUA)
-	require.True(t, ok)
-
-	require.Len(t, o.RootNodes, 2)
-	require.Equal(t, o.RootNodes[0].FieldName, "name")
-	require.Equal(t, o.RootNodes[1].FieldName, "name2")
-
-	require.Len(t, o.Groups, 2)
-	require.Equal(t, o.Groups[0].MetricName, "foo")
-	require.Len(t, o.Groups[0].Nodes, 1)
-	require.Equal(t, o.Groups[0].Nodes[0].Identifier, "3000")
-
-	require.NoError(t, o.InitNodes())
-	require.Len(t, o.nodes, 4)
-	require.Len(t, o.nodes[0].metricTags, 1)
-	require.Len(t, o.nodes[1].metricTags, 2)
-	require.Len(t, o.nodes[2].metricTags, 3)
-	require.Len(t, o.nodes[3].metricTags, 2)
-
-	require.Len(t, o.Workarounds.AdditionalValidStatusCodes, 1)
-	require.Equal(t, o.Workarounds.AdditionalValidStatusCodes[0], "0xC0")
-}
 
-func TestConfigWithMismatchedTypes(t *testing.T) {
-	toml := `
-[[inputs.opcua]]
-name = "localhost"
-endpoint = "opc.tcp://localhost:4840"
-connect_timeout = "10s"
-request_timeout = "5s"
-security_policy = "auto"
-security_mode = "auto"
-certificate = "/etc/telegraf/cert.pem"
-private_key = "/etc/telegraf/key.pem"
-auth_method = "Anonymous"
-username = ""
-password = ""
-nodes = [
-  {name="name", namespace="1", identifier_type="s", identifier="one"},
-  {name="name2", namespace="2", identifier_type="i", identifier="two"},
-]
```

Comment:
did we lose this test case?

Reply:
It moved into `inputs/common/opcua/input_client/input_client_test.go`